### PR TITLE
RFE #23 - Add new status [WARNING] - Check: is system self registered

### PR DESCRIPTION
**Pull request by swapab**
- Add check `Checks::SystemRegistration`
- Add new status and label named `warning`
- Add context block `test/minitest_spec_context.rb`

### CODE REVIEW

File: lib/minitest_spec_context.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'minitest/spec'
+
+module MiniTest
```
Comment (iNecas) [inline lib/minitest_spec_context.rb]: Please keep this in `test/test_helper.rb` or some file in that directory. It looks strange having it in `lib` which should not contain test-related code

**Comment by iNecas** [comment]
One comment, but looks good otherwise

File: lib/minitest_spec_context.rb (hunk 1 above)
Comment (swapab) [inline lib/minitest_spec_context.rb]: :+1:
Moved it to `test/helpers/minitest_spec_context.rb`

File: definitions/checks/system_registration.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      raise ForemanMaintain::Error::Warn, 'System is self registered'
+    else
+      puts 'System is not self registered'
+    end
+  end
+
+  private
+
+  def system_is_self_registerd?
+    rhsm_hostname.include?(hostname)
```
Comment (iNecas) [inline definitions/checks/system_registration.rb]: We should match only the last word from the line and match on equality, rather than include: you can have `sat.example.com' registered to `another-sat.example.com` and this check would fail: a corner case, but still…

File: test/helpers/minitest_spec_context.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'minitest/spec'
```
Comment (iNecas) [inline test/helpers/minitest_spec_context.rb]: we use `support` directory in `test/lib`, we can name this directory `support` as well, for consistency

Comment (swapab) [inline test/helpers/minitest_spec_context.rb]: Yes I had thought of that. I thought it would be an `ambiguity` :stuck_out_tongue:

File: definitions/checks/system_registration.rb (hunk 2 above)
Comment (swapab) [inline definitions/checks/system_registration.rb]: Aah. Good catch. I did a split on ` = ` later checked for `hostname` in the derived array.

File: definitions/checks/system_registration.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      raise ForemanMaintain::Error::Warn, 'System is self registered'
+    else
+      puts 'System is not self registered'
+    end
+  end
+
+  private
+
+  def system_is_self_registerd?
+    rhsm_hostname.split(' = ').include?(hostname)
```
Comment (iNecas) [inline definitions/checks/system_registration.rb]: No need to match for the string before ' = ', also, there could be 'hostname=myhost.example.com' that would not get though this. This (or similar variant) should be better `rhsm_hostname.sub(/.*?=/,'').strip`

Comment (swapab) [inline definitions/checks/system_registration.rb]: Better :+1: Thanks!

File: definitions/checks/system_registration.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      puts 'System is not self registered'
+    end
+  end
+
+  def system_is_self_registerd?
+    rhsm_hostname.casecmp(hostname).zero?
+  end
+
+  def rhsm_hostname
+    execute("grep '\\bhostname\\b' < /etc/rhsm/rhsm.conf").sub(/.*?=/, '').strip
```
Comment (iNecas) [inline definitions/checks/system_registration.rb]: While testing, I hit sill one issue, when I had a line commented out with this value. `'^\\s*\\bhostname\\b'` to match online uncommented line.

Comment (swapab) [inline definitions/checks/system_registration.rb]: Handled this case as well. Thanks!

**Comment by iNecas** [comment]
Thanks @swapab
